code/modules/clothing/suits/utility.dm

```diff
@@ -47,27 +47,76 @@
 	name = "bomb hood"
 	desc = "Use in case of bomb."
 	icon_state = "bombsuit"
-	flags = HEADCOVERSEYES|HEADCOVERSMOUTH|BLOCKHAIR
-	armor = list(melee = 0, bullet = 0, laser = 0,energy = 0, bomb = 100, bio = 0, rad = 0)
+	w_class = 5//Too large to fit in a backpack
+	flags = HEADCOVERSEYES|HEADCOVERSMOUTH|BLOCKHAIR|STOPPRESSUREDAMAGE|THICKMATERIAL|BLOCK_GAS_SMOKE_EFFECT
+	armor = list(melee = 30, bullet = 20, laser = 25,energy = 30, bomb = 100, bio = 60, rad = 60)
 	flags_inv = HIDEMASK|HIDEEARS|HIDEEYES
 	body_parts_covered = HEAD|FACE|EYES
 	siemens_coefficient = 0
 
 
+
+//Changes by Nanako
+//Bomb suits should huge and robust. They used to have 100 bomb protection and nothing in other categories
+//Bomb suits now have decent resistance in all categories, but with two major ergonomic drawbacks:
+//1. Heavy. really heavy, massive slowdown on movement
+//2. Bomb suit materials don't allow permeability of body heat, thus the wearer tends to overheat and can't wear them for long
 /obj/item/clothing/suit/bomb_suit
 	name = "bomb suit"
-	desc = "A suit designed for safety when handling explosives."
+	desc = "A suit designed for safety when handling explosives. It looks heavy and uncomfortable to wear for even a short time."
 	icon_state = "bombsuit"
 	item_state = "bombsuit"
-	w_class = 4//bulky item
+	w_class = 7//bulky item
 	gas_transfer_coefficient = 0.01
 	permeability_coefficient = 0.01
-	slowdown = 2
-	armor = list(melee = 0, bullet = 0, laser = 0,energy = 0, bomb = 100, bio = 0, rad = 0)
+	slowdown = 7
+	armor = list(melee = 50, bullet = 50, laser = 40,energy = 30, bomb = 100, bio = 60, rad = 60)
+	flags = STOPPRESSUREDAMAGE|THICKMATERIAL
 	flags_inv = HIDEJUMPSUIT|HIDETAIL
-	heat_protection = UPPER_TORSO|LOWER_TORSO
-	max_heat_protection_temperature = ARMOR_MAX_HEAT_PROTECTION_TEMPERATURE
+	heat_protection = UPPER_TORSO|LOWER_TORSO|LEGS|FEET|ARMS
+	cold_protection = UPPER_TORSO|LOWER_TORSO|LEGS|FEET|ARMS
+	max_heat_protection_temperature = SPACE_SUIT_MAX_HEAT_PROTECTION_TEMPERATURE
 	siemens_coefficient = 0
+	var/mob/living/carbon/human/wearer = null
+	var/suit_temp = T20C
+
+/obj/item/clothing/suit/bomb_suit/equipped(var/mob/user, var/slot)
+	if (slot == slot_wear_suit && !(src in processing_objects))
+		wearer = user
+		wearer << "<span class='Notice'>You struggle into the [src]. It feels hot, heavy and uncomfortable</span>"
+		processing_objects.Add(src)
+
+
+	..(user, slot)
+
+#define 	BOMBSUIT_THERMAL	0.27
+#define		BOMBHOOD_THERMAL	0.12
+#define		BOMBSUIT_MAX_TEMPERATURE	420	//heat 2 for humans, heat 1 for unathi
+/obj/item/clothing/suit/bomb_suit/process()
+	if (!wearer || !istype(loc, /mob))//If nobody's wearing the suit, then it cools down
+		suit_temp -= 0.5
+		if (suit_temp < T20C)
+			suit_temp = T20C
+			processing_objects.Remove(src)
+		return
+
+	var/amount = BOMBSUIT_THERMAL
+	if (istype(wearer.head, /obj/item/clothing/head/bomb_hood))//wearing both parts heats up faster

```

Comment:
Won't this generate a runtime log in the instance where the suit is carried in hand (at which point, istype(loc, /mob) == TRUE but wearer is still null)?